### PR TITLE
Publish request function in plugin pipeline and query string parameters on URI

### PR DESCRIPTION
**Pull request by Alx101**
- adds publish_request to plugin pipeline
- adds query member to URI object with appropriate manipulation and parsing

### CODE REVIEW

**Comment by Alx101** [comment]
Some tests fail due to missing python packages:
- https://travis-ci.org/5monkeys/content-io/jobs/635172369?utm_medium=notification&utm_source=github_status
- https://travis-ci.org/5monkeys/content-io/jobs/635172373?utm_medium=notification&utm_source=github_status

**Comment by Alx101** [comment]
Dropped tox checks and support for python versions 3.3, 3.4 and 2.7 by recommendation from @lundberg. Should probably add support for 3.7 and 3.8 later on.

File: cio/api.py [hunk 1]
```diff
@@ -102,18 +104,24 @@ def uri_chain(uri):
...
-        plugin = plugins.resolve(node['uri'])  # May raise UnknownPlugin and should be handled outside api
-        data = node['data'] = plugin.load(source)
-        node['content'] = plugin.render(data)
+        plugin = plugins.resolve(node.uri)  # May raise UnknownPlugin and should be handled outside api
+        data = plugin._load(node)
+        node.content = plugin._render(data, node)
+        # if node.get_content_stack_count() > 1:
+        #     data = node.get_content_at(-2)
+        # else:
+        #     data = node.content
```
Comment (lundberg) [inline cio/api.py]: Remove commented code

File: cio/node.py [hunk 2]
```diff
@@ -49,6 +49,12 @@ def set_uri(self, uri):
 
     uri = property(get_uri, set_uri)
 
+    def get_content_at(self, index):
+        return self._content[index]
+
+    def get_content_stack_count(self):
+        return len(self._content)
+
```
Comment (lundberg) [inline cio/node.py]: Remove, if not used.

File: cio/plugins/base.py [hunk 3]
```diff
@@ -18,20 +18,51 @@ def load(self, content):
...
         return data
 
+    def _save(self, node):
+        """
+        Perform action on node, persist external plugin resources and return content string for plugin data
+        """
+        node.content = self.save(node.content)
+        return node
+
+    def publish(self, node):
```
Comment (lundberg) [inline cio/plugins/base.py]: Rename to `_publish` to align with other methods taking `node` arg.

**Pull request merged**